### PR TITLE
ResourceFilterStoreService

### PR DESCRIPTION
**Pull request by randallmorey**
This PR introduces the `ResourceFilterStoreService` to more easily apply resource filters to an API query.  Example usage:

```js
    const filter = new ResourceFilter({
      type: 'oidc',
      status: ['active', 'pending']
    });
    console.log(filter.queryExpression);
    // ("/item/type" == "oidc") and ("/item/status" == "active" or "/item/status" == "pending")
```

### CODE REVIEW

File: addons/api/addon/services/resource-filter-store.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+   * Similar to Ember's built-in `store.query`, except this method accepts
+   * an additional argument `filterObject` representing a query filter
+   * (see ResourceFilter class above).
+   * @param {string} modelName
+   * @param {object} filterObject
+   * @param {?object} storeQuery
+   * @return {Promise}
+   */
+  queryBy(modelName, filterObject = {}, storeQuery = {}) {
+    console.log(filterObject);
```
Comment (DhariniJeeva) [inline addons/api/addon/services/resource-filter-store.js:103]:
```suggestion
```

File: addons/api/addon/services/resource-filter-store.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+  // =methods
+
+  /**
+   * Similar to Ember's built-in `store.query`, except this method accepts
+   * an additional argument `filterObject` representing a query filter
+   * (see ResourceFilter class above).
+   * @param {string} modelName
+   * @param {object} filterObject
+   * @param {?object} storeQuery
```
Comment (DhariniJeeva) [inline addons/api/addon/services/resource-filter-store.js:99]: is `?object` a typo?

**Comment by DhariniJeeva** [comment]
LGTM!!

**Review by DhariniJeeva — APPROVED** [review]

File: addons/api/addon/services/resource-filter-store.js (hunk 2 above)
Comment (randallmorey) [reply addons/api/addon/services/resource-filter-store.js:99]: Intentional.  This indicates that the argument is optional.  Technically both objects are optional, but the whole point of this service is to pass a filter object.

**Pull request merged**